tests/ValidationTest.php

```diff
@@ -103,5 +103,31 @@ public function testExists(): void
             ['name' => 'required|exists:users']
         );
         $this->assertFalse($validator->fails());
+
+        $validator = Validator::make(
+            ['name' => ['test name', 'john']], // Part of an existing value
+            ['name' => 'required|exists:users']
+        );
+        $this->assertTrue($validator->fails());
```

Comment:
Just to confirm, `$validator->fails()` is expected to return `true` here because we're checking that a user with that exact name exists?

Reply:
Yes, according to the previous tests, the validation must fail when the value is **not** found in the collection.